web/pdf_link_service.js

```diff
@@ -258,9 +258,15 @@ var PDFLinkService = (function PDFLinkServiceClosure() {
             mode: params.pagemode
           });
         }
-      } else if (isPageNumber(hash)) { // Page number.
-        this.page = hash | 0;
       } else { // Named (or explicit) destination.
+        if ((typeof PDFJSDev === 'undefined' || PDFJSDev.test('GENERIC')) &&
+            (isPageNumber(hash) && hash < this.pagesCount)) {
```

Comment:
D'oh! You're obviously correct, don't know how I managed to make that kind of mistake in such a small patch. Good eye, thanks :-)